rerun_py/src/remote.rs

```diff
@@ -226,48 +205,33 @@ impl PyStorageNodeClient {
         })
     }
 
-    /// Update the metadata for the recording with the given id.
+    /// Update the catalog metadata for one or more recordings.
+    ///
+    /// The updates are provided as a pyarrow Table or RecordBatch containing the metadata to update.
+    /// The Table must contain an 'id' column, which is used to specify the recording to update for each row.
     ///
     /// Parameters
     /// ----------
-    /// id : str
-    ///     The id of the recording to update.
-    /// metadata : dict[str, MetadataLike]
-    ///     A dictionary where the keys are the metadata columns and the values are pyarrow arrays.
+    /// metadata : Table | RecordBatch
+    ///     A pyarrow Table or RecordBatch containing the metadata to update.
     #[pyo3(signature = (
-        id,
         metadata
     ))]
-    fn update_catalog(&mut self, id: &str, metadata: &Bound<'_, PyDict>) -> PyResult<()> {
+    #[allow(clippy::needless_pass_by_value)]
+    fn update_catalog(&mut self, metadata: MetadataLike) -> PyResult<()> {
         self.runtime.block_on(async {
-            let (schema, data): (
-                Vec<arrow2::datatypes::Field>,
-                Vec<Box<dyn arrow2::array::Array>>,
-            ) = metadata
-                .iter()
-                .map(|(key, value)| {
-                    let key = key.to_string();
-                    let value = value.extract::<MetadataLike>()?;
-                    let value_array = value.to_arrow2()?;
-                    let field =
-                        arrow2::datatypes::Field::new(key, value_array.data_type().clone(), true);
-                    Ok((field, value_array))
-                })
-                .collect::<PyResult<Vec<_>>>()?
-                .into_iter()
-                .unzip();
+            let metadata = metadata.into_record_batch()?;
 
-            let schema = arrow2::datatypes::Schema::from(schema);
-
-            let data = arrow2::chunk::Chunk::new(data);
+            // TODO(jleibs): This id name should probably come from `re_protos`
```

Comment:
(I was thinking about this yesterday and came to the same conclusion. We should put these rerun mandatory fields somewhere so that we can re-use on both sides)